### PR TITLE
Feature/alexzhang1618/dashboard dropdown

### LINKED ISSUE #57: Frontend: Dashboard dropdown functionality

### PR DESCRIPTION
**Pull request by alexzhang1618**
### Administrative Info
[GitHub Issue #57]
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Implemented functionality for group selector dropdown in Dashboard.js
- Add and style next and previous page buttons in Table.js and Table.css
- Selected group defaults to first group in dropdown if available
- Page selector disappears on tables with 0 rows

### Testing
How did you confirm your changes worked? 
- Ran the React app and interacted with the elements

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Group Selector Dropdown |
| --- |
| Screen Shot 2022-03-10 at 11 07 03 AM |

| Page Selector Buttons |
| --- |
| Screen Shot 2022-03-10 at 11 07 44 AM |

### CODE REVIEW

**Review by eliasfang — APPROVED** [review]
Looks and works great! Nice work!

File: frontend/src/components/Table.js [hunk 1]
```diff
@@ -6,51 +6,67 @@
...
-          ))}
-        </tr>
-        {data.map((entry) => (
-          <tr className="table-body-row" key={entry._id}>
-            {_schema.map((field) => (
-              <td className="table-body-cell">{entry.data[field]}</td>
+        <tbody>
+          <tr className="table-header-row">
+            {group.Values.map((field) => (
+              <th className="table-header-cell">{field.name}</th>
```
Comment (wllmwu) [inline frontend/src/components/Table.js:33]: Should have a `key` prop on the `th` elements. I'm not sure if field names are required to be unique--if so then those can be the keys, if not then just the index I guess

If you use the index, then you can also add `/* eslint-disable react/no-array-index-key */` above that part of the code to disable the lint warning

File: frontend/src/components/Table.js [hunk 2]
```diff
@@ -6,51 +6,67 @@
...
             ))}
-            <img src={Pencil} className="pencil-svg" alt="edit icon on table row" />
           </tr>
-        ))}
+          {data
+            .slice((currentPage - 1) * elementsPerPage, currentPage * elementsPerPage)
+            .map((entry) => (
+              <tr className="table-body-row" key={entry._id}>
+                {group.Values.map((field) => (
+                  <td className="table-body-cell">{entry.data[field.name]}</td>
```
Comment (wllmwu) [inline frontend/src/components/Table.js:41]: Add a `key` prop on the `td`s too

**Review by wllmwu — COMMENTED** [review]
Nice. These things below don't break the app currently, although there's a small chance they'll cause bugs in the future. They do create a lot of errors in the console though. Also we've both changed Dashboard.js so there will be conflicts :( @navidboloorian what should we do about this

File: frontend/src/components/Table.js (hunk 2 above)
Comment (navidboloorian) [reply frontend/src/components/Table.js]: I've gone ahead and added the names of the field as a key to resolve the error. We can change this in the future if need be.

**Review by navidboloorian — APPROVED** [review]
Good stuff. Fixed some syntax errors. I'll merge.

**Pull request merged**